### ISSUE TITLE
this.target is undefined when clearing search results

### ISSUE THREAD

**Issue by astab** [jamalneufeld/ghostHunter#47]
Hi there,

I am getting the following error when trying to clear search results and search field value with searchField.clear(); :

Uncaught TypeError: Cannot read property 'val' of undefined
at Object.clear (jquery.ghosthunter.js?v=79caedeb82:3532)
at HTMLButtonElement. (custom.js?v=79caedeb82:87)
at HTMLButtonElement.dispatch (jquery-3.3.1.min.js:2)
at HTMLButtonElement.y.handle (jquery-3.3.1.min.js:2)

Could you please help? Thank you.

**Comment by Windyo** [comment]
I have tried and failed to reproduce this this weekend.
I am assuming you are using the latest version of ghost, from your other posts.
Anything particular about your current setup ?

**Comment by astab** [comment]
Ghost version: 1.21.3
jQuery: 3.3.1

There is nothing particular about the set up (Digital Ocean one-click install). You can see the demo here: http://192.241.152.207/ . The search input field value should be cleared when the search close button is clicked:

```
var searchField = $('#search-field').ghostHunter({
	results : "#search-results",
	onKeyUp : true,
	onPageLoad : true,
	includepages : true,
	info_template : '<div class="results-info">Posts found: {{amount}}</div>',
	result_template : '<a href="{{link}}" id="gh-{{ref}}" class="result-item gh-search-item"><div class="result-title">{{title}}</div><div class="result-date">{{pubDate}}</div></a>'
});
$('.search-toggle').on('click', function(e){
	var $body = $('body');
	if ( $body.hasClass('search--opened') ) {
		$body.removeClass('search--opened');
		searchField.clear();
	} else {
		$body.addClass('search--opened');
		setTimeout(function() {
			$('#search-field').focus();
		}, 300);
	}
	e.preventDefault();
});
```

**Comment by Windyo** [comment]
Strangely enough it's working on mine. Checking what I did differently...

Just to test something though, can you search for 
```this.target.val("");```
in your js and replace it with
```target.val("");```

?

**Comment by astab** [comment]
Just tried that, didn't help.

**Comment by TaoK** [comment]
I had the same thing, discovered when upgrading ghosthunter in my customized copy of the "polar" theme.

The "very slightly broken" ghosthunter file I used was dist/jquery.ghosthunter.js on master (as of today).

The problem was that the "clear" plugin function referenced "this.target", but the "target" property was never being saved into the plugin object, only referenced from function argument in the "init" function.

I fixed it by adding a line "that.target = target;" as the second line in the init function, thereby keeping the clearable element around in the plugin as expected.

No pull request, sorry, I haven't set up gulp etc, I'm hacking away at this very unprofessionally at the moment.
